graphql/resolvers.go

```diff
@@ -288,6 +290,83 @@ func (r *mutationResolver) SpawnHost(ctx context.Context, spawnHostInput *SpawnH
 	return &apiHost, nil
 }
 
+func (r *mutationResolver) EditSpawnHost(ctx context.Context, editSpawnHostInput *EditSpawnHostInput) (*restModel.APIHost, error) {
+	usr := route.MustHaveUser(ctx)
+	h, err := host.FindOneByIdOrTag(editSpawnHostInput.HostID)
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error finding host by id: %s", err))
+	}
+
+	if !CanUpdateSpawnHost(h, usr) {
+		return nil, Forbidden.Send(ctx, "You are not authorized to modify this host")
+	}
+
+	if editSpawnHostInput.DisplayName != nil {
+		err = h.SetDisplayName(*editSpawnHostInput.DisplayName)
+		if err != nil {
+			return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error while modifying spawnhost name: %s", err))
+		}
+	}
+	if editSpawnHostInput.NoExpiration != nil {
+		if *editSpawnHostInput.NoExpiration {
+			h.MarkShouldNotExpire(ExpireInDays(evergreen.SpawnHostExpireDays))
+		} else {
+			h.MarkShouldExpire(ExpireInDays(evergreen.SpawnHostExpireDays))
+		}
+	}
+	if editSpawnHostInput.Expiration != nil {
+		err = h.SetExpirationTime(*editSpawnHostInput.Expiration)
+		if err != nil {
+			return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error while modifying spawnhost expiration time: %s", err))
+		}
+	}
+	if editSpawnHostInput.InstanceType != nil {
+		config, err := evergreen.GetConfig()
+		if err != nil {
+			return nil, InternalServerError.Send(ctx, "unable to retrieve server config")
+		}
+		foundInstanceType := false
+		for _, a := range config.Providers.AWS.AllowedInstanceTypes {
```

Comment:
Instead of doing this I think you want to call cloud.CheckInstanceTypeValid instead, which additionally checks with the provider that this is valid